test/clj/rems/api/test_resources.clj

```diff
@@ -91,46 +90,62 @@
             (json-body {:resid resid
                         :organization "TEST-ORGANIZATION"
                         :licenses [licid]})
-            handler
+            (@handler)
             assert-response-is-ok)
         (testing "and fetch"
           (let [data (-> (request :get "/api/resources")
                          (authenticate api-key user-id)
-                         handler
+                         (@handler)
                          assert-response-is-ok
                          read-body)
                 resource (some #(when (= resid (:resid %)) %) data)]
             (is resource)
-            (is (= [licid] (map :id (:licenses resource))))))
-        (testing "duplicate resource ID is not allowed within one organization"
-          (let [response (-> (request :post "/api/resources/create")
-                             (authenticate api-key user-id)
-                             (json-body {:resid resid
-                                         :organization "TEST-ORGANIZATION"
-                                         :licenses [licid]})
-                             handler)
-                body (read-body response)]
-            (is (= 200 (:status response)))
-            (is (= false (:success body)))
-            (is (= [{:type "t.administration.errors/duplicate-resid" :resid resid}] (:errors body)))))
-        (testing "duplicate resource ID is allowed between organizations"
-          (-> (request :post "/api/resources/create")
-              (authenticate api-key user-id)
-              (json-body {:resid resid
-                          :organization "TEST-ORGANIZATION2"
-                          :licenses [licid]})
-              handler
-              assert-response-is-ok))))))
+            (is (= [licid] (map :id (:licenses resource))))))))))
+
+(deftest resources-api-duplicate-resid-test
+  (let [api-key "42"
+        user-id "owner"
+        licid 1
+        resid "RESOURCE-API-DUPLICATE-TEST"]
+    (-> (request :post "/api/resources/create")
+        (authenticate api-key user-id)
+        (json-body {:resid resid
+                    :organization "TEST-ORGANIZATION"
+                    :licenses [licid]})
+        (@handler)
+        assert-response-is-ok)
+    (testing "duplicate resource ID is allowed between organizations"
+      (-> (request :post "/api/resources/create")
+          (authenticate api-key user-id)
+          (json-body {:resid resid
+                      :organization "TEST-ORGANIZATION2"
+                      :licenses [licid]})
+          (@handler)
+          assert-response-is-ok))
```

Comment:
The API returns HTTP 200 also when there are errors, so this test is not asserting anything. It should check that the response has `:success true` (both here and the request right above it).

Reply:
Ok this is old code so you should fix it in another PR. This PR was only a proof of concept that doesn't make sense to merge but only to keep as discussion.

Reply:
#1109 created